lib/private/help-exec-machine-instance.js

```diff
@@ -806,7 +806,7 @@ module.exports = function helpExecMachineInstance (liveMachine) {
       } catch (e) {
 
         // Split the error we got into lines, and remove the first one (the actual error message)
-        e.stack = e.stack.split('\n');
+        e.stack = stripStackMessage(e).split('\n');
         var msg = e.stack.shift();
```

Comment:
At this point in the code, `e.stack` is actually a trace string _without_ the message included, right?  So I don't think that `msg` will actually be the message here (to get the message, we can do `e.message`).

So, taking the other comments into account, this becomes:

```js
// Grab the bare trace from the Error instance (i.e. `.stack` without the message prefix),
// then split that bare trace string on newlines to get an array.
var newStackEntries = getBareTraceFromError(e).split('\n');

// Strip stack entries that originate from the internals of the machine runner.
newStackEntries = removeNonUserlandCallsFromStack(newStackEntries);

// Now combine the pruned stack entries stack entries from our new error with the 
// with the pre-existing stack entries saved in this machine's rolling `env`, smashing
// it all together into a new stack trace string.
var newBareTrace = getCombinedErrorStack(
  newStackEntries,
  liveMachine._configuredEnvironment.debug.stack
);

// Then save that new trace string back onto our Error instance (overwriting the original trace string).
// Notice that we also prepend the conventional prefix: the error name and message, with punctuation.
e.stack = e.name + ': ' + e.message + '\n' + newBareTrace;

// Also set an additional `stackModified` property to indicate that we've already massaged
// this Error's stack, and that we won't need to do it again.
Object.defineProperty(e, 'stackModified', {value: true});
```

Note that the above code takes into account that we removed line 818 below, in favor of only modifying `e.stack` in one place.